_posts/2019/2019-12-01-ensure-you-correctly-build-caching-keys.md

```diff
@@ -0,0 +1,110 @@
+---
+title: "Ensure you correctly build your caching keys"
+description: "Why won't this model update? Oh."
+layout: article
+category: ruby
+image:
+  base: '2019/ensure-you-correctly-build-caching-keys'
+  alt: 'Keys'
+  credit:  Chunlea Ju
+  source: "https://unsplash.com/photos/8fs1X0JFgFE"
+
+---
+
+Caching is a hugely powerful tool in maintaining performance of often requested pages and partials in your application.
+
+It can also be a cause of confusing behaviours (mostly “stuff not updating on the page”) when you test your app in a production environment.
+
+Rails has great built-in support for view fragment caching. The framework includes an elegant way of invoking and busting the cache based on the model `id`, it’s `updated_at` timestamp and an automatically generated digest of the specific view template. You can find more details in the [Rails guide on caching](https://guides.rubyonrails.org/caching_with_rails.html#fragment-caching).
+
+
+## Instead of...
+
+...just using the modal key in your key:
+
+```ruby
+class Event < ApplicationRecord
+  has_many :attendees
+end
+
+class Attendee < ApplicationRecord
+  belongs_to :event
+  has_many :prerequistes
+end
+
+class Prerequistes < ApplicationRecord
+  belongs_to :attendee
+
+  scope :done, -> { where.not(completed_at: nil) }
+end
+```
+
+
+### `app/views/events/show.html.erb`
+
+```erb
+<% @event.attendees.each do |attendee| %>
+  <% cache [attendee] do %>
+    <p>
+      <%= attendee.name %>, <%= @event.name %><br />
+      <small>
+        <%= attendee.prerequistes.done.count %> done
+      </small>
+    </p>
+  <% end %>
+<% end %>
+```
+
+
+## Use
+
+...a combined key, including the parent object
```

Comment:
Colon or fullstop at end of this.

And at this point I'm still not sure exactly what we're trying to guard against/achieve here. Maybe the setup/opening narrative needs to be a bit clearer.

Reply:
Maybe part of the problem is the sudden intro of the word 'key' here?